pkg/daemon/ceph/config/config.go

```diff
@@ -189,7 +200,22 @@ func CreateDefaultCephConfig(context *clusterd.Context, cluster *ClusterInfo, ru
 	i := 0
 	for _, monitor := range cluster.Monitors {
 		monMembers[i] = monitor.Name
-		monHosts[i] = monitor.Endpoint
+		monIP := cephutil.GetIPFromEndpoint(monitor.Endpoint)
+
+		// This tries to detect the current port is the mon already exist
+		// This basically handles the transtion between monitors running on 6790 to msgr2
+		currentMonPort := cephutil.GetPortFromEndpoint(monitor.Endpoint)
+
+		monPorts := [2]string{"3300", currentMonPort}
+		newEndPoint := net.JoinHostPort(monIP, monPorts[0])
+		oldEndPoint := net.JoinHostPort(monIP, monPorts[1])
+
+		cephVersion := os.Getenv("ROOK_CEPH_VERSION_NAME")
```

Comment:
I think it is more transparent to pass in the ceph version from the calling function than to read an environment variable.

Reply:
I'm not sure if you've had to work with the `rook/cmd` dir yet; it's something I found confusing at first. I think adding a string variable named `"ceph-version-name"` in `rook/cmd/rook/ceph` (probably in `ceph.go`) would follow the existing patterns. 

There is some metamagic going on in the background of the env vars. The value can be set either by commandline flag in the daemon `spec.go` files (`--ceph-version-name`), or it can be loaded automagically from the `ROOK_CEPH_VERSION_ENVIRONMENT` variable.

Reply:
I guess for this to work completely I need your followup on configuring osd/rgw/nfs etc config from the operator.

Reply:
Actually nop, my bad.